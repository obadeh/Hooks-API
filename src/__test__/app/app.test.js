/* eslint-disable no-unused-vars */
import React from 'react';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';

configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../../app.js';

describe('<App />', () => {

  it('exists at the start of the application', () => {
    let app = shallow(<App />);
    expect(app.find('div').exists()).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});