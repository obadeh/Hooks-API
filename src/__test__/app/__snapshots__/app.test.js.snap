// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <header>
    <h2>
      <p>
        There are
      </p>
      0
      " Items To Complete
    </h2>
  </header>
  <section
    className="todo"
  >
    <div>
      <h3>
        Add Item
      </h3>
      <form
        onSubmit={[Function]}
      >
        <label>
          <span>
            To Do Item
          </span>
          <input
            name="text"
            onChange={[Function]}
            placeholder="Add To Do List Item"
          />
        </label>
        <label>
          <span>
            Difficulty Rating
          </span>
          <input
            defaultValue="3"
            max="5"
            min="1"
            name="difficulty"
            onChange={[Function]}
            type="range"
          />
        </label>
        <label>
          <span>
            Assigned To
          </span>
          <input
            name="assignee"
            onChange={[Function]}
            placeholder="Assigned To"
            type="text"
          />
        </label>
        <label>
          <span>
            Due
          </span>
          <input
            name="due"
            onChange={[Function]}
            type="date"
          />
        </label>
        <button>
          Add Item
        </button>
      </form>
    </div>
    <div>
      <ul />
      <button
        onClick={[Function]}
      >
        Hide completed
      </button>
      <button
        onClick={[Function]}
      >
        Show completed
      </button>
    </div>
  </section>
  <footer>
    <p>
      23:18:22 GMT+0200 (Eastern European Standard Time)
    </p>
    <h3>
      By obada Q 2020
    </h3>
  </footer>
</div>
`;
